pkg/destroy/powervs/powervs.go

```diff
@@ -0,0 +1,956 @@
+
+	if strings.HasPrefix(config.IAMAccessToken, "Bearer") {
+		bluemixToken = config.IAMAccessToken[7:len(config.IAMAccessToken)]
+	} else {
+		bluemixToken = config.IAMAccessToken
+	}
+
+	token, err := jwt.Parse(bluemixToken, func(token *jwt.Token) (interface{}, error) {
+		return "", nil
+	})
+	if err != nil && !strings.Contains(err.Error(), "key is of invalid type") {
+		return &user, err
+	}
+
+	claims := token.Claims.(jwt.MapClaims)
+	if email, ok := claims["email"]; ok {
+		user.Email = email.(string)
+	}
+	user.ID = claims["id"].(string)
+	user.Account = claims["account"].(map[string]interface{})["bss"].(string)
+	iss := claims["iss"].(string)
+	if strings.Contains(iss, "https://iam.cloud.ibm.com") {
+		user.cloudName = "bluemix"
+	} else {
+		user.cloudName = "staging"
+	}
+	user.cloudType = "public"
+
+	user.generation = generation
+	return &user, nil
+}
+
+// GetRegion converts from a zone into a region.
+func GetRegion(zone string) (region string, err error) {
+	err = nil
+	switch {
+	case strings.HasPrefix(zone, "dal"), strings.HasPrefix(zone, "us-south"):
+		region = "us-south"
+	case strings.HasPrefix(zone, "sao"):
+		region = "sao"
+	case strings.HasPrefix(zone, "us-east"):
+		region = "us-east"
+	case strings.HasPrefix(zone, "tor"):
+		region = "tor"
+	case strings.HasPrefix(zone, "eu-de-"):
+		region = "eu-de"
+	case strings.HasPrefix(zone, "lon"):
+		region = "lon"
+	case strings.HasPrefix(zone, "syd"):
+		region = "syd"
+	case strings.HasPrefix(zone, "tok"):
+		region = "tok"
+	case strings.HasPrefix(zone, "osa"):
+		region = "osa"
+	case strings.HasPrefix(zone, "mon"):
+		region = "mon"
+	default:
+		return "", fmt.Errorf("region not found for the zone: %s", zone)
+	}
+	return
+}
+
+// ClusterUninstaller holds the various options for the cluster we want to delete.
+type ClusterUninstaller struct {
+	APIKey         string
+	BaseDomain     string
+	CISInstanceCRN string
+	ClusterName    string
+	Context        context.Context
+	InfraID        string
+	Logger         logrus.FieldLogger
+	Region         string
+	ServiceGUID    string
+	VPCRegion      string
+	Zone           string
+
+	managementSvc  *resourcemanagerv2.ResourceManagerV2
+	controllerSvc  *resourcecontrollerv2.ResourceControllerV2
+	vpcSvc         *vpcv1.VpcV1
+	zonesSvc       *zonesv1.ZonesV1
+	dnsRecordsSvc  *dnsrecordsv1.DnsRecordsV1
+	piSession      *ibmpisession.IBMPISession
+	instanceClient *instance.IBMPIInstanceClient
+	imageClient    *instance.IBMPIImageClient
+	jobClient      *instance.IBMPIJobClient
+	keyClient      *instance.IBMPIKeyClient
+
+	resourceGroupID string
+	cosInstanceID   string
+
+	errorTracker
+	pendingItemTracker
+}
+
+// New returns an IBMCloud destroyer from ClusterMetadata.
+func New(logger logrus.FieldLogger, metadata *types.ClusterMetadata) (providers.Destroyer, error) {
+	var apiKey string
+
+	apiKey = os.Getenv("IBMCLOUD_API_KEY")
+	if apiKey == "" {
+		return nil, errors.New("environment variable IBMCLOUD_API_KEY must be set")
+	}
```

Comment:
Same comment as above. We should change this pattern not to rely on environment variables.

Reply:
Fixed